streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableTransformValuesTest.java

```diff
@@ -163,104 +159,74 @@ public void shouldNotSendOldValuesByDefault() {
         final Processor<String, Change<String>, String, Change<String>> processor = transformValues.get();
         processor.init(context);
 
-        context.forward(new Record<>("Key", new Change<>("Key->newValue!", null), 0));
-        expectLastCall();
-        replay(context);
+        doNothing().when(context).forward(new Record<>("Key", new Change<>("Key->newValue!", null), 0));
 
         processor.process(new Record<>("Key", new Change<>("newValue", "oldValue"), 0));
-
-        verify(context);
     }
 
     @Test
     public void shouldSendOldValuesIfConfigured() {
         final KTableTransformValues<String, String, String> transformValues =
             new KTableTransformValues<>(parent, new ExclamationValueTransformerSupplier(), null);
 
-        expect(parent.enableSendingOldValues(true)).andReturn(true);
-        replay(parent);
+        when(parent.enableSendingOldValues(true)).thenReturn(true);
 
         transformValues.enableSendingOldValues(true);
         final Processor<String, Change<String>, String, Change<String>> processor = transformValues.get();
         processor.init(context);
 
-        context.forward(new Record<>("Key", new Change<>("Key->newValue!", "Key->oldValue!"), 0));
-        expectLastCall();
-        replay(context);
+        doNothing().when(context).forward(new Record<>("Key", new Change<>("Key->newValue!", "Key->oldValue!"), 0));
 
         processor.process(new Record<>("Key", new Change<>("newValue", "oldValue"), 0));
-
-        verify(context);
-    }
-
-    @Test
-    public void shouldNotSetSendOldValuesOnParentIfMaterialized() {
```

Comment:
This test checks nothing as far as I can tell because the QUERYABLE_NAME isn't equal to null, which is when we would reach the if-condition to throw the exception at.

There is another test which tests the behaviour when that value is null, but it doesn't throw an exception, so given that this is a deprecated I decided to not try to cover this scenario.

I have also not used verifyNoMoreInteractions, but if reviewers think it is necessary I can circle back and add it.

Reply:
I don't want to make calls on removing tests while reviewing this PR because now as a reviewer I have to look at the code & understand what is going, why test can be removed etc. instead of simply focusing on ensuring that easymock -> mockito transformations are correct. 

In the spirit of keeping PR reviewer friendly, could we add this test back?

Reply:
That's what I am trying to explain, however - this test is incorrect and the act of moving it from EasyMock to Mockito surfaced this problem. This means that either this test needs to be removed or it needs to be changed.

Reply:
Correct me if I am wrong here but this test validates that code never executes:
```
if (parent.enableSendingOldValues(forceMaterialization)) {
            sendOldValues = true;
        }
```
in `enableSendingOldValues()`. The way how this test does it is by throwing an exception on it's call and if the code executes, the exception will be thrown and test will fail.

You can change it by explicitly verifying that no calls to `parent.enableSendingOldValues` is made in this code.

Reply:
You are entirely correct!

Mockito was reporting that that exception was never thrown, but I completely missed checking the message in the exception which would have given me the needed information to change the verification, apologies.